backend/app/repository/github_provider.py

```
@@ -256,15 +259,48 @@ async def get_branches(
-                    default=branch.get("default", False)
+                    default=branch.get("default", False) or branch["name"] == default_branch_name
                 ).model_dump() for branch in all_branches
             ]
         except requests.exceptions.RequestException as e:
             raise HTTPException(
                 status_code=502,
                 detail=f"GitHub API error: {str(e)}"
             )
-    
+            
+    @lru_cache(maxsize=100)
```

Ruff (0.14.2)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)